tsflex/features/feature_collection.py

```diff
@@ -367,7 +367,7 @@ def calculate(
         # determing the bounds of the series dict items and slice on them
         start, end = _determine_bounds(bound_method, list(series_dict.values()))
         series_dict = {
-            n: s[s.index.dtype.type(start) : s.index.dtype.type(end)]
+            n: s.loc[s.index.dtype.type(start) : s.index.dtype.type(end)]  # TODO: check memory efficiency of ths
```

Comment:
Check memory efficiency of this

Reply:
Runtime is the same as previous implementation :heavy_check_mark: 
![image](https://user-images.githubusercontent.com/18898740/164993510-e729551c-13b4-4d1b-bc43-413e678adaf9.png)

Reply:
Memory profiling indicates no memory peak :heavy_check_mark: 
![image](https://user-images.githubusercontent.com/18898740/164994239-c5a63ab6-ca93-4f39-860a-e9dfdfe0162a.png)